src/tests/ftest/container/snapshot_functionality_test.py

```diff
@@ -0,0 +1,539 @@
+                , self.container)
+
+        #(4)Test snapshot with a NULL container handler
+        self.log.info("==(4)Snapshot with a NULL container handler.")
+        if self.invalid_snapshot_test(None, epoch):
+            self.log.info("==>Negative test 2, expecting failed on taking "
+                          "snapshot on a NULL container.coh.")
+        else:
+            self.fail("##(4)Negative test 2 passing, expecting failed on "
+                      "taking snapshot with a NULL container.coh.")
+
+        #(5)Test snapshot with an invalid epoch
+        self.log.info("==(5)Snapshot with a NULL epoch.")
+        if self.invalid_snapshot_test(self.container.coh, None):
+            self.log.info("==>Negative test 3, expecting failed on taking "
+                          "snapshot with a NULL epoch.")
+        else:
+            self.fail("##(5)Negative test 3 passing, expecting failed on "
+                      "taking snapshot with a NULL epoch.")
+
+
+    def test_snapshot_info(self):
+        """
+        Test ID: DAOS-1386 Test container SnapShot information
+                 DAOS-1371 Test list snapshots
+                 DAOS-1395 Test snapshot destroy
+                 DAOS-1402 Test creating multiple snapshots
+        Test Description:
+                (1)Create an object, write random data into it, and take
+                   a snapshot.
+                (2)Make changes to the data object. The write_an_obj function
+                   does a commit when the update is complete.
+                (3)Verify the data in the snapshot is the original data.
+                   Get a handle for the snapshot and read the object at dkey,
+                   akey. Compare it to the originally written data.
+                (4)List the snapshot and make sure it reflects the original
+                   epoch.
+                   ==>Repeat step(1) to step(4) for multiple snapshot tests.
+                (5)Verify the snapshots data.
+                (6)Destroy the snapshot.
+                (7)Check if still able to Open the destroyed snapshot and
+                   Verify the snapshot removed from the snapshot list.
+        Use Cases: Combinations with minimun 1 client and 1 server.
+        :avocado: tags=snap,snapshotinfo
+        """
+
+        coh_list = []
+        container_epoch_list = []
+        snapshot_list = []
+        test_data = []
+        snapshot_index = 1
+        snapshot_loop = 10
+        obj_cls = self.params.get("obj_class", '/run/object_class/*')
+        akey = self.params.get("akey", '/run/snapshot/*', default="akey")
+        dkey = self.params.get("dkey", '/run/snapshot/*', default="dkey")
+        rand_str = lambda n: ''.join([random.choice(string.lowercase)
+                                      for i in xrange(n)])
+        #
+        #Test loop for creat, modify and snapshot object in the DAOS container.
+        #
+        while snapshot_index <= snapshot_loop:
```

Comment:
sorry, i won't understand what this snapshot loop is about. Could you please elaborate?